code/obj/retribution/makeshift_signaller.dm

```diff
@@ -0,0 +1,168 @@
+var/sword_summoned_before = false
+
+/obj/item/makeshift_signaller_frame
+	name = "makeshift signaller frame"
+	icon = 'icons/misc/retribution/makeshift_signaller.dmi'
+	icon_state = "frame"
+	flags = FPRINT | TABLEPASS | CONDUCT | ONBELT
+	w_class = 2.0
+	throw_speed = 4
+	throw_range = 20
+	m_amt = 500
+	burn_type = 1
+	var/build_stage = 0
+	mats = 4
+	desc = "A disemboweled remote signaller, ready for further modifications."
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 1
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (build_stage >= 3)											//If build_stage is 3 or higher, which shouldn't be possible, alert the player to in turt alert coders.
+			user.show_message("<span class='notice'>Uh oh, it seems you broke it!</span>", 1)
+			desc = "This doodad is broken. Call a coder."
+			tooltip_rebuild = 1
+			return
+		else if (ispulsingtool(W) && build_stage <= 0)					//Step 1 of construction: Multitool.
+			build_stage = 1
+			user.show_message("<span class='notice'>You rearrange and attune the wiring inside!</span>", 1)
+			desc = "A remote signaller frame with wiring sticking out."
+			tooltip_rebuild = 1
+			return
+		else if (istype(W,/obj/item/sheet) && build_stage == 1)			//Step 2 of construction: Metal.
+			W.amount -= 1
+			if (W.amount <= 0)
+				qdel(W)
+			else
+				W.inventory_counter.update_number(W.amount)
+			build_stage = 2
+			user.show_message("<span class='notice'>You make a crude but functional circuit board port and slot it into the frame!</span>", 1)
```

Comment:
You can make a generic circuit board in general manufactors, you could probably use that instead of plain metal sheets. It'd make way more sense and would give the item some proper use

Reply:
Hm, didn't know that. I agree, that sounds better, thanks. Will change either today or tomorrow.

Reply:
I'm gonna make it use a Secure Data Circuit Board, rather than a normal one. That way Security personnel will also know that something's up, if they see the Detective's laptop broken at least.

Reply:
Done.